ProductsService/ProductsService.Persistence/Clients/ReadDatabaseMongoClient.cs

```diff
@@ -0,0 +1,21 @@
+﻿using Microsoft.Extensions.Configuration;
+using MongoDB.Driver;
+
+namespace ProductsService.Persistence.Clients
+{
+    internal class ReadDatabaseMongoClient
+    {
+        private readonly IMongoDatabase Database;
+        private readonly MongoClient MongoClient;
+
+        public ReadDatabaseMongoClient(IConfiguration configuration)
+        {
+            MongoClient = new MongoClient(configuration["ReadMongoSettings:Connection"]);
+
+            Database = MongoClient.GetDatabase(configuration["ReadMongoSettings:DatabaseName"]);
```

Comment:
this can be moved to configuration and you can inject Database in context constructor

Reply:
I can't because I have two instances of the client and database.

Reply:
then inject to constructor configured mongo settings